### PR TITLE
Use `target` instead of `machine` for mir interpreter integer handling.

### PR DESCRIPTION
**Pull request by oli-obk**
The naming of `machine` only makes sense from a mir interpreter internals perspective, but outside users talk about the `target` platform. As per https://github.com/rust-lang/rust/pull/108029#issuecomment-1429791015

r? @RalfJung

### CODE REVIEW

File: compiler/rustc_const_eval/src/const_eval/target.rs [hunk 1]
```diff
@@ -0,0 +1,656 @@
+use rustc_hir::def::DefKind;
```
Comment (RalfJung) [inline compiler/rustc_const_eval/src/const_eval/target.rs:1]: What happened here, why is there such a new file?

Comment (oli-obk) [reply compiler/rustc_const_eval/src/const_eval/target.rs]: huh... that was in my edits during branch switching and probably got saved by vscode after the branch switch removed it

**Comment by RalfJung** [comment]
So this renames machine_usize → target_usize but leaves other uses of the term "machine" unchanged? LGTM, r=me with CI happy.

**Comment by oli-obk** [comment]
> leaves other uses of the term "machine" unchanged

yea. I looked at the other uses, like the trait and modules, but it's not really about the target. The `Machine` trait impls for example are different mir interpreters and, while internally handling target specific things, that's not their purpose. There are no more `fn .*machine.*` except those that refer to LLVM or GCC specific backend logic.

**Comment by RalfJung** [comment]
> The Machine trait impls 

Those are about the "Abstract Machine" and I think the name still makes sense.

**Pull request merged**